frontend/src/App.tsx

```diff
@@ -83,28 +81,43 @@ const App = () => {
    * Handler function for 'datasource' select element.
    * Updates dataSource with the values picked by the user in Sidebar.
    */
-  const onSelectDataSource = (event) => {
-    const dataSourceValues = event.map((item) => item.value);
+  const onSelectDataSource = (
+    value: OptionsType<{
+      value: string;
+      label: string;
+    }>,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    actionMeta: ActionMeta<{
+      value: string;
+      label: string;
+    }>
+  ) => {
+    const dataSourceValues = value.map((item: SelectOption) => item.value);
```

Comment:
```suggestion
    const dataSourceValues = value.map((item) => item.value);
```

don't think you need the type in there